ibis/common/annotations.py

```diff
@@ -193,6 +193,22 @@ class Signature(inspect.Signature):
 
     @classmethod
     def merge(cls, *signatures, **annotations):
+        """Merge multiple signatures.
+
+        In addition to concatenating the parameters, it also reorders the
+        parameters so that optional arguments come after mandatory arguments.
+
+        Parameters
+        ----------
+        *signatures : Signature
+            Signature instances to merge.
+        **annotations : dict
```

Comment:
```suggestion
        signatures
            Signature instances to merge.
        annotations
```

The `*`/`**` are not valid parameter names, and the types should be annotations, which will get picked up by `mkdocstrings` when rendering docstrings.